### PR TITLE
cq-editor: fix build

### LINKED ISSUE #91483: cq-editor fails to build in nixpkgs-unstable
**Describe the bug**
the cq-editor package fails to build with an error message that indicates a dependency (cadquery) does not work on python 3.8.

```
λ nix-shell -p cq-editor
error: cadquery-2.0 not supported for interpreter python3.8
(use '--show-trace' to show detailed location information)
```

cadquery itself has no issues being used:

```
λ nix-shell -p python37Packages.cadquery
these paths will be fetched (34.45 MiB download, 324.73 MiB unpacked):
  /nix/store/gc0lw2sz5wc567vslbfz8py2i8shg6g1-python3.7-cadquery-2.0
  /nix/store/hyz7s6274x58hdbzb7slrjdg4knq4l24-pythonocc-core-cadquery-0.18.2
copying path '/nix/store/hyz7s6274x58hdbzb7slrjdg4knq4l24-pythonocc-core-cadquery-0.18.2' from 'https://cache.nixos.org'...
copying path '/nix/store/gc0lw2sz5wc567vslbfz8py2i8shg6g1-python3.7-cadquery-2.0' from 'https://cache.nixos.org'...

[nix-shell:~/engineering/cadquery-nix]$
```

**To Reproduce**
1. Type into the terminal `nix-shell -p cq-editor`

**Expected behavior**
I expected cq-editor to available in the resulting nix-shell.

**Notify maintainers**
@costrouc  @marcus7070

**Metadata**

```
λ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/0gmrcxw1bmbinqi3hszmmy62qiqrvpmy-nix-info
copying path '/nix/store/0gmrcxw1bmbinqi3hszmmy62qiqrvpmy-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.37, NixOS, 20.09pre228204.467ce5a9f45 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.5`
 - channels(root): `"nixos-20.09pre231796.22a81aa5fc1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by marcus7070**
###### Motivation for this change

Will close #91483 and do the `spyder_3` half of #91484 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Unfortunately I can't get cq-editor working with Python 3.8, so I've set it to 3.7.

@jonringer, I see you added toml to pylint's nativeBuildInputs in https://github.com/NixOS/nixpkgs/commit/27f0dabfbfbc15a4233827f08fa36e725f27ca54. I think it needs to be propagated due to the errors in #91484 and https://github.com/PyCQA/pylint/blob/a73585740398052ccd6c67d33869d9049a27e8c7/pylint/__pkginfo__.py#L43 so I've done that in this PR. Is that OK?

### CODE REVIEW

File: pkgs/applications/graphics/cq-editor/default.nix [hunk 1]
```diff
@@ -1,10 +1,10 @@
 { lib
 , mkDerivationWith
-, python3Packages
```
Comment (FRidh) [inline pkgs/applications/graphics/cq-editor/default.nix:3]: please make the change in `all-packages.nix`, `python3Packages = python37Packages;`.

Comment (marcus7070) [reply pkgs/applications/graphics/cq-editor/default.nix]: Done, thankyou @FRidh

**Comment by marcus7070** [comment]
@jonringer, are you able to have a look at the pylint change?

**Review by jonringer — APPROVED** [review]

**Comment by jonringer** [comment]
> @jonringer, I see you added toml to pylint's nativeBuildInputs in [27f0dab](https://github.com/NixOS/nixpkgs/commit/27f0dabfbfbc15a4233827f08fa36e725f27ca54). I think it needs to be propagated due to the errors in #91484 and https://github.com/PyCQA/pylint/blob/a73585740398052ccd6c67d33869d9049a27e8c7/pylint/__pkginfo__.py#L43 so I've done that in this PR. Is that OK?

Yea, this is more correct.

Thought the toml package was needed just for reading a `pyproject.toml` or something during setup

**Comment by jonringer** [comment]
this fixes two previously broken builds LGTM

**Pull request merged**